Enable display of github links on dataset page and add a test scenario

rija opened the pull request:
# Pull request for issue: #855

This is a pull request for the following functionalities:

Github links not displayed on dataset page under the external links section


## How to test?

Navigate to ``http://gigadb.gigasciencejournal.com:9170/dataset/100935``

You should see a section for Github links and a link to a GitHub repo below	

## How have functionalities been implemented?

In ``protected/views/dataset/view.php``

I have added the "Github Links" external type name to the list of external link types to display as parameter to the ``getDatasetExternalLinksTypesAndCount()`` method.

## Any issues with implementation?

none

## Any changes to automated tests?

Added an acceptance test scenario for the implemented functionality in ``tests/acceptance/DatasetView.feature``.
Also fixed an erroneous link in the project image scenario (it's correct on develop already , but there was a mishap during a past rebase) 

Added a new docker compose service for modernised Selenium/WebDriver functionality to work with Apple silicon based Mac. 
Added the same specific IP address to the ``chrome`` and ``chrome-arm`` docker service  and configure the acceptance tests to use that IP address irrespective of which of the two service is actually launched by ``up.sh``.
up.sh is the one that will detect the architecture and choose one over the other based on that.

All tests runners should pass whether we are running on Intel or Apple Silicon based Mac.
Relevant changes are in:

* ops/deployment/docker-compose.yml
* up.sh
* tests/acceptance_runner
* tests/acceptance.suite.yml
* behat.yml

To test that, executing all runners should work as usual:
```
$ ./tests/unit_runner && ./tests/functional_runner && ./tests/unit_functional_runner
$ ./tests/acceptance_runner local 
```

rija committed:
Enable display of github links on dataset page and add a test scenario

rija committed:
Minor tweak

rija committed:
Fix a rebasing error

rija committed:
Allow acceptance_runner to run on apple silicon based mac

rija committed:
Change approach for getting acceptance tests running on apple silicon

the previous approach didn't work on Intel based mac,

so instead of replacing the docker compose service for webdriver, I've add an additional one
specific to arm and kept the old one for intel based

then updated the up.sh script to select which one to instantiate based on current architecture.

Also reverted composer.lock and composer.json.dist to the version from develop for now. We will verify whether that need to change

rija committed:
Fix error that got two properties mixed up

rija committed:
run Codeceoption acceptance tests first

pli888 commented on tests/acceptance/DatasetView.feature:
```diff
@@ -73,3 +73,10 @@ Feature: a user visit the dataset page
     Then I should see an image "https://assets.gigadb-cdn.net/live/images/projects/genome_10k/G10Klogo.jpg" is linked to "http://www.genome10k.org/"
     And I should see an image "https://assets.gigadb-cdn.net/live/images/projects/the_avian_phylogenomic_project/phylogenomiclogo.png" is linked to "http://avian.genomics.cn/en/index.html"
+
+@ok
```
Indent `@ok` tag

pli888 reviewed (approved):
The code is working for me as I can see the Github link at http://gigadb.gigasciencejournal.com:9170/dataset/100935. All of the unit, functional and acceptance tests are passing locally on my Intel Mac too, including the new acceptance test that checks Github links are displayed on dataset pages.

My only query is that the updates to the 2 composer.lock files weren't reported in the pull request.

kencho51 reviewed (approved):
Hi @rija,

All tests are passing locally and in my gitlab CI/CD pipeline. And when I go to `http://gigadb.gigasciencejournal.com:9170/dataset/100935`, the github link is available.
Other than Peter's comments, the code looks good to me, so I would like to approve it.

rija committed:
Fix indentation of @ok tag in feature file

rija committed:
Restore develop version of file-worker composer.lock

as I haven't make any change that should change it

rija commented:
Hi @pli888, @kencho51,

Thanks for reviewing and spotting those issues. Here's a follow up:

* I've now fixed the tag indentation in the feature file
* I've now reverted the ``gigadb/app/worker/file-worker/composer.lock`` file to its ``develop`` version as I've done no change that should elicit its update

Regarding the changes to ``composer.lock``, I will keep them as they were supposed to be part of the merged PR #1025 but I forgot to add them then.

If there are no further comments, I'll merge this PR shortly.
